tests/integration/relations/opensearch_provider/application-charm/src/charm.py

```diff
@@ -183,27 +261,28 @@ def request(
         if endpoint.startswith("/"):
             endpoint = endpoint[1:]
 
-        # add username and password if auth continues to fail
         full_url = f"https://{host}:{port}/{endpoint}"
+
+        request_kwargs = {
+            "verify": False,  # TODO this should be a cert once this relation has TLS.
+            "method": method.upper(),
+            "url": full_url,
+            "headers": {"Content-Type": "application/json"},
+        }
+
+        if payload:
+            if isinstance(payload, dict):
+                payload = json.dumps(payload)
+            request_kwargs["data"] = payload
+            request_kwargs["headers"]["Accept"] = "application/json"
```

Comment:
The `Accept` header is related to the expected content type of the response from the server, whereas `Content-Type` is related to the sort of content the client sends to the server. 
I think this is unnecessary and it's pretty safe to just put it as part of the `headers` dict above